core/http/src/uri/authority.rs

```diff
@@ -109,6 +123,25 @@ impl<'a> Authority<'a> {
         crate::parse::uri::authority_from_str(string)
     }
 
+    /// Parses the string `string` into an `Authority`. Parsing will never
```

Comment:
This is good. Can we make the doc string match that of `Origin` in the latest commit?